plugins/wacom-usb/fu-wac-device.c

```diff
@@ -726,6 +725,58 @@ fu_wac_device_add_modules(FuWacDevice *self, GError **error)
 		return FALSE;
 	}
 
+	/* validate versions of each submodule */
+	for (guint8 i = 0; i < buf[3]; i++) {
+		guint8 fw_type = buf[(i * 4) + 4] & ~0x80;
+		guint16 ver;
+
+		if (!fu_memread_uint16_safe(buf,
+					    sizeof(buf),
+					    (i * 4) + 5,
+					    &ver,
+					    G_BIG_ENDIAN,
+					    error))
+			return FALSE;
+
+		/* module is in transitional state or requires re-flashing */
+		if (ver == 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "module %u has error state",
+				    fw_type);
+			return FALSE;
+		}
+	}
+
+	memcpy(out, buf, sizeof(buf));
+	return TRUE;
+}
+
+static gboolean
+fu_wac_device_add_modules(FuWacDevice *self, GError **error)
+{
+	g_autofree gchar *version_bootloader = NULL;
+	guint8 buf[32] = {0xff};
+	guint16 boot_ver;
+	g_autoptr(GError) error_local = NULL;
+
+	/* wait for all modules started successfully */
+	if (!fu_device_retry_full(FU_DEVICE(self),
+				  fu_wac_device_add_modules_cb,
+				  FU_WAC_DEVICE_MODULE_RETRY_DELAY,
+				  FU_WAC_DEVICE_TIMEOUT / FU_WAC_DEVICE_MODULE_RETRY_DELAY,
+				  buf,
+				  &error_local)) {
+		if (error_local->code != FWUPD_ERROR_NOT_FOUND)
+			return FALSE;
```

Comment:
this bit isn't valid -- you need to `g_propagate_error(error, g_steal_pointer(&error_local))` before returning.

Reply:
oh! thanks!
Strange but nothing failed then I simulate that path.

Reply:
I would have expected SIGSEG -- maybe explore a bit and work out why! You'll find another bug I'm sure. :)